app/src/main/java/tech/ula/ui/FilesystemListFragment.kt

```diff
@@ -46,6 +51,27 @@ class FilesystemListFragment : Fragment() {
         }
     }
 
+    private val filesystemExportStatusObserver = Observer<FilesystemExportStatus> {
+        it?.let { exportStatus ->
+            when (exportStatus) {
+                is ExportUpdate -> {
+                    activityContext.updateExportProgress(exportStatus.details)
+                }
+                is ExportSuccess -> {
+                    Toast.makeText(activityContext, R.string.export_success, Toast.LENGTH_LONG).show()
+                    activityContext.stopExportProgress()
+                }
+                is ExportFailure -> {
+                    Toast.makeText(activityContext, "${R.string.export_failure} + ${exportStatus.reason}", Toast.LENGTH_LONG).show()
```

Comment:
Does `${R.string.export_failure} + ${exportStatus.reason}` work as expected? Seems to me like it would like just call `toString` on the `Int` itself. This could be fixed by either using string formatting in the string resource itself, or by calling `getString` on the resource and prepending the result to `exportStatus.reason`.

Reply:
Additionally, I think failure modes could be better served by a dialog than a toast. That way, the user will have a better chance to understand the failure as well as report it to us.